src/Cipher/AutoKeyString.ts

```diff
@@ -0,0 +1,65 @@
+import CaesarUtils from './CaesarUtils';
+
+class AutoKeyString {
+  private _text: string;
+  private _key: string;
+
+  constructor(text: string = '', key: string = '') {
+    if (key.length < 1) {
+      throw new Error('A key is required');
```

Comment:
Ugh.  This can't throw and work with puzzle-web, but it's legitimately meaningless to try to "decrypt" an autokey without a key (example: A ciphertext of N, without key, could have been either G or T).  I'll have to special case an empty key, but I don't have time to do that this morning quite yet.

Reply:
Same at line 28

Reply:
Yeah, in general it's probably best to just accept empty values and do the right thing.